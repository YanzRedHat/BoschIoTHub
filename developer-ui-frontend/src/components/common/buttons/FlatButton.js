/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Button from "./Button";
import { getThemeColor } from "utils";
// SVG Imports
import SubmitIcon from "images/submitCaret.svg";

const FlatBtn = Button.extend`
  -webkit-backface-visibility: hidden;
  border: 0;
  outline: none;
  background: none;
  opacity: 0.65;
  transition: opacity 0.18s ease-out;
  color: ${props => getThemeColor(props)};
  display: inline-flex;
  align-items: center;
  justify-content: center;

  a {
    text-decoration: none;
    color: inherit;
  }
  svg {
    path {
      ${props =>
        props.submitAnimation
          ? `fill: ${props.theme.disabledFontColorHiContrast};`
          : `fill: ${getThemeColor(props)};`};
    }
  }

  &:hover {
    opacity: 1;
    ${props =>
      props.submitAnimation &&
      `
    svg {
      transform: translateX(0.5em);
      path {
        fill: ${getThemeColor(props)};
      }
    }`};
  }
  &:disabled {
    color: ${props => props.theme.disabledFontColorHiContrast};
    &:hover {
      svg {
        transform: translateX(0);
        path {
          transform: translateX(0);
          fill: ${props => props.theme.disabledFontColorHiContrast};
        }
      }
    }
  }
`;

const SubmitSvg = styled(SubmitIcon)`
  transition: transform 0.3s ease-out;
  padding-right: 0.75em;
  height: 0.6em;
  path {
    transition: fill 0.5s ease-out;
  }
`;

const FlatButton = props => {
  const {
    children,
    submitAnimation,
    primary,
    secondary,
    danger,
    ...other
  } = props;
  return (
    <FlatBtn
      primary={primary ? 1 : 0}
      secondary={secondary ? 1 : 0}
      danger={danger ? 1 : 0}
      submitAnimation={submitAnimation}
      {...other}>
      {submitAnimation && <SubmitSvg {...other} />}
      {children}
    </FlatBtn>
  );
};

FlatButton.propTypes = {
  type: PropTypes.string,
  children: PropTypes.string,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  danger: PropTypes.bool,
  submitAnimation: PropTypes.bool
};

export default FlatButton;
