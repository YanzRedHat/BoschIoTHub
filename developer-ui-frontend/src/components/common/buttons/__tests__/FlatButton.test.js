/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import React from "react";
import { shallow } from "enzyme";
import FlatButton from "components/common/buttons/FlatButton.js";
import CopyIcon from "images/copyClipboardIcon.svg";

describe("FlatButton", () => {
  it("should be defined", () => {
    expect(FlatButton).toBeDefined();
  });
  it("should render correctly", () => {
    const tree = shallow(<FlatButton name="button test" />);
    expect(tree).toMatchSnapshot();
  });
  it("should have a prop primary", () => {
    const tree = shallow(<FlatButton name="button test" primary />);
    expect(tree.props().primary).toEqual(1);
  });
  it("should have a prop secondary", () => {
    const tree = shallow(<FlatButton name="button test" secondary />);
    expect(tree.props().secondary).toEqual(1);
  });
  it("should have a prop danger", () => {
    const tree = shallow(<FlatButton name="button test" danger />);
    expect(tree.props().danger).toEqual(1);
  });
  it("should have a prop submitAnimation", () => {
    const tree = shallow(<FlatButton name="button test" submitAnimation />);
    expect(tree.props().submitAnimation).toEqual(true);
  });
  it("should have a prop type", () => {
    const tree = shallow(<FlatButton name="button test" type="submit" />);
    expect(tree.props().type).toEqual("submit");
  });
  it("should call mock function when button is clicked", () => {
    const mockFn = jest.fn();
    const tree = shallow(<FlatButton name="button test" onClick={mockFn} />);
    tree.simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });
  it.skip("Compared values have no visual difference but may have different properties", () => {
    const tree = shallow(<FlatButton name="button test" children={CopyIcon} />);
    expect(JSON.stringify(tree.props().children)).toEqual(
      JSON.stringify(<span>{CopyIcon} </span>)
    );
  });
});
