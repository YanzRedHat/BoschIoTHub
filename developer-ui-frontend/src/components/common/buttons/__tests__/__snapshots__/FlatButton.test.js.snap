// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatButton should render correctly 1`] = `
<styled.button
  danger={0}
  name="button test"
  primary={0}
  secondary={0}
/>
`;
